src/core/session.c

```diff
@@ -189,6 +192,12 @@ MsQuicSessionOpen(
         goto Error;
     }
 
+    if (Settings != NULL &&
+        SettingsSize < (uint32_t)FIELD_OFFSET(QUIC_SETTINGS, MaxBytesPerKey)) {
```

Comment:
why not `sizeof`?

Reply:
Because I don't want this size to change when we add a new setting.

Reply:
Is the goal that newer MsQuic versions can work with clients that are targeting an older version?

Reply:
Yes. The scenario is app A compiles and ships against MsQuic 1.0.0. We then ship MsQuic 1.1.0 (or 1.0.1) which is guaranteed not to be a breaking change; and therefor the app should be able to drop in the new msquic.dll (or .so) without updating the app. In order to support adding new settings not being a breaking change, this check here only requires the original size. Anything beyond this is optional.